packages/superset-ui-chart/src/components/ChartDataProvider.tsx

```diff
@@ -0,0 +1,147 @@
+/* eslint react/sort-comp: 'off' */
+import React, { ReactNode } from 'react';
+import { SupersetClientInterface, RequestConfig } from '../../../superset-ui-connection/src/types';
+
+import ChartClient, { SliceIdAndOrFormData } from '../clients/ChartClient';
+import { ChartFormData } from '../types/ChartFormData';
+import { QueryData } from '../models/ChartProps';
+
+type Datasource = object;
```

Comment:
Do we want to be more specific here?

Reply:
`ChartClient` never really defined a specific `Datasource` type so I was just using that logic. But I agree it'd be useful so I updated it to be more specific in `types/Datasource` (erring on the side of under-defining for now, but using the seemingly most-important fields from an example API response)

Reply:
ok